annubes/tasks/task.py

```diff
@@ -11,74 +11,86 @@
             defined in `session`, followed by catch trials.
         max_sequential: Maximum number of sequential trials of the same modality. Only used if shuffle is True.
             Defaults to None (no maximum).
-        n_outputs: Number of output nodes in the network, signaling different behavioral choices.
-            Defaults to 2.
-        output_behavior: List of possible intensity values of the behavioral output. Currently only the smallest and
-            largest value of this list are used.
-            Defaults to [0, 1].
-        stim_time: Duration of each stimulus in ms.
-            Defaults to 1000.
         fix_intensity: Intensity of input signal during fixation.
             Defaults to 0.
         fix_time: Fixation time in ms. Note that the duration of each input and output signal is increased by this time.
             Defaults to 100.
-        input_baseline: Baseline input for all neurons.
-            Defaults to 0.2.
         iti: Inter-trial interval, or time window between sequential trials, in ms.
             Defaults to 0.
         dt: Sampling interval (inverted sampling frequency) in ms.
             Defaults to 20.
         tau: Time constant for the dynamics of each network node in ms.
             Defaults to 100.
-        rescaling_coeff: Rescaling coefficient for `self.stim_intensities` and `self.fix_intensity`. If set to non-zero
-            value, these values are linearly rescaled along (0, rescaling_coeff).
-            Defaults to 0 (i.e. no rescaling).
+        n_outputs: Number of output nodes in the network, signaling different behavioral choices.
+            Defaults to 2.
+        output_behavior: List of possible intensity values of the behavioral output. Currently only the smallest and
+            largest value of this list are used.
+            Defaults to [0, 1].
+        input_baseline: Baseline input for all neurons.
+            Defaults to 0.2.
         noise_std: Standard deviation of input noise.
             Defaults to 0.01.
-
-    Raises:
-        ValueError: if `catch_prob` is not between 0 and 1.
+        rescaling_coeff: Rescaling coefficient for `Task.stim_intensities` and `self.fix_intensity`. If set to non-zero
+            value, these values are linearly rescaled along (0, rescaling_coeff).
+            Defaults to 0 (i.e. no rescaling).
     """
 
-    name: str
-    session: dict[str, float] = field(default_factory=lambda: {"v": 0.5, "a": 0.5})
-    stim_intensities: list[float] = field(default_factory=lambda: [0.8, 0.9, 1])
-    catch_prob: float = 0.5
     shuffle_trials: bool = True
     max_sequential: int | None = None
-    n_outputs: int = 2
-    output_behavior: list[float] = field(default_factory=lambda: [0, 1])
-    stim_time: int = 1000
     fix_intensity: float = 0
     fix_time: int | None = 100
-    input_baseline: float = 0.2
     iti: int = 0
```

Comment:
I think `fix_time` and `iti` should not allow None inputs, but just int (where None is essentially 0 ms).

These two attributes are dealt with similarly throughout the code, when they are checked for `if not None`, even though `iti` doesn't allow for `None` input according to the type hint. I don't know what that check does that wouldn't be covered by setting them to 0, which would mean 0 timepoints receive the associated value.